ansible/roles/ssc/templates/ssc-deploy.yml.j2

```diff
@@ -0,0 +1,131 @@
+{% endif %}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ssc
+  namespace: {{ ssc_namespace }}
+  labels:
+    app: ssc
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ssc
+  template:
+    metadata:
+      labels:
+        app: ssc
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "{{ ssc_metrics_port }}"
+    spec:
+      initContainers:
+        - name: init-data
+          image: busybox
+          command: ['sh', '-c', 'if [ ! -f /data/priv_validator_state.json ]; then echo {"height": "0", "round": 0, "step": 0} > /data/priv_validator_state.json; fi']
+          volumeMounts:
+            - name: ssc-data
+              mountPath: /data
+        - name: init-validator
+          image: {{ ssc_image }}
+          command:
+            - sh
+            - -c
+            - |
+              if [ ! -f /data/config/priv_validator_key.json ]; then
+                echo "Initializing validator..."
+                sscd init {{ ssc_key_name }} --home /data --chain-id {{ ssc.chain_id }} --default-denom {{ ssc.denom }}
+                
+                echo "Adding validator account..."
+                echo "$VALIDATOR_MNEMONIC" | sscd keys add {{ ssc_key_name }} --home /data --keyring-backend test --recover --no-backup
+                
+                echo "Adding genesis account..."
+                sscd add-genesis-account {{ ssc_key_name }} 1000000000000{{ ssc.denom }} --home /data --keyring-backend test
+                
+                echo "Creating gentx..."
+                sscd genesis gentx {{ ssc_key_name }} 1000000000{{ ssc.denom }} --home /data --keyring-backend test --chain-id {{ ssc.chain_id }}
+                
+                echo "Collecting gentxs..."
+                sscd genesis collect-gentxs --home /data
+                
+                echo "Validator initialization complete"
+              else
+                echo "Validator already initialized"
+              fi
+          env:
+            - name: KEY_PASSWORD
+              value: "{{ ssc.key_password }}"
+            - name: VALIDATOR_MNEMONIC
+              value: "{{ validator_mnemonic }}"
+          volumeMounts:
+            - name: ssc-data
+              mountPath: /data
+      containers:
+        - name: ssc
+          image: {{ ssc_image }}
+          command: ["sscd", "start", "--home", "/data", "--rpc.laddr", "tcp://0.0.0.0:{{ ssc_rpc_port }}", "--p2p.laddr", "tcp://0.0.0.0:{{ ssc_p2p_port }}", "--grpc.address", "0.0.0.0:{{ ssc_grpc_port }}"]
+          ports:
+            - containerPort: {{ ssc_rpc_port }}
+              name: rpc
+            - containerPort: {{ ssc_p2p_port }}
+              name: p2p
+            - containerPort: {{ ssc_grpc_port }}
+              name: grpc
+            - containerPort: {{ ssc_metrics_port }}
+              name: metrics
+          volumeMounts:
+            - name: ssc-data
+              mountPath: /data
```

Comment:
Data folder should be mounted under `~/.ssc/data`. But there is a bigger problem here, the start script you wrote requires persistent config folder, which I don't see here.